### PR TITLE
[DDW-1076] Incorrect amount of tokens is sent

### PR DESCRIPTION
**Pull request by renanvalentin**
This PR fixes race condition issue on send screen form. 

## Todos

- [ ] TODO

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1649961602048609)
- [ ] Test

---

### Basics

- [ ] PR assigned to the PR author(s)
- [ ] `input-output-hk/daedalus-dev` and `input-output-hk/daedalus-qa` assigned as PR reviewers
- [ ] If there are UI changes, Alexander Rukin assigned as an additional reviewer
- [ ] All visual regression testing has been reviewed (assign `run Chromatic` label to PR to trigger the run)
- [ ] PR has appropriate labels (`release-vNext`, `feature`/`bug`/`chore`, `WIP`)
- [ ] PR link is added to a Jira ticket, ticket moved to In Review
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes
- [ ] PR contains screenshots (in case of UI changes)
- [ ] CHANGELOG entry has been added to the top of the appropriate section (_Features_, _Fixes_, _Chores_) and is linked to the correct PR on GitHub
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality

- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with typescript types
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing

- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review

- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Review by lucas-barros — APPROVED** [review]

**Review by shawnbusuttil — APPROVED** [review]

**Review by marcin-mazurek — CHANGES_REQUESTED** [review]

**Review by gabriela-ponce — APPROVED** [review]
Nice job 🎉

File: source/renderer/app/components/wallet/WalletSendForm.spec.tsx [hunk 1]
```diff
@@ -415,4 +452,88 @@ describe('wallet/Wallet Send Form', () => {
     assertAdaInput(minimumAda);
     assertMinimumAmountNoticeMessage(minimumAda);
   });
+
+  test('should wait fees to be calculated before submitting', async () => {
+    expect.assertions(3);
+
+    const mock = jest
```
[the remark this answers is not in the document]
Comment (marcin-mazurek) [reply source/renderer/app/components/wallet/WalletSendForm.spec.tsx]: sounds good

File: source/renderer/app/components/wallet/WalletSendForm.spec.tsx [hunk 2]
```diff
@@ -21,6 +22,34 @@ import { BrowserLocalStorageBridge } from '../../features/local-storage';
 import { HwDeviceStatuses } from '../../domains/Wallet';
 import WalletTokenPicker from './tokens/wallet-token-picker/WalletTokenPicker';
 import WalletSendForm from './WalletSendForm';
+import { FORM_VALIDATION_DEBOUNCE_WAIT } from '../../config/timingConfig';
+
+jest.mock(
```
[the remark this answers is not in the document]
Comment (marcin-mazurek) [reply source/renderer/app/components/wallet/WalletSendForm.spec.tsx:25]: Makes sense - not ideal but I think fixing this properly would increase the scope of this bug fix significantly.

File: source/renderer/app/components/wallet/WalletSendForm.spec.tsx [hunk 3]
```diff
@@ -415,4 +468,153 @@ describe('wallet/Wallet Send Form', () => {
...
+        );
+
+        expect(adaAmountConfirmation).toHaveTextContent(expectedAdaAmount);
+
+        const totalAmountConfirmation = screen.getByTestId(
+          'confirmation-dialog-total-amount',
+          {}
+        );
+
+        expect(totalAmountConfirmation).toHaveTextContent(expectedTotalAmount);
```
Comment (marcin-mazurek) [inline source/renderer/app/components/wallet/WalletSendForm.spec.tsx:613]: The formating seems to be broken?

**Review by marcin-mazurek — APPROVED** [review]

**Pull request merged**